app/src/main/java/com/tubitv/media/demo/UI/PlayerControllerUI.java

```diff
@@ -0,0 +1,101 @@
+package com.tubitv.media.demo.UI;
+
+import android.content.Context;
+import android.databinding.DataBindingUtil;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.FrameLayout;
+import com.tubitv.demo.R;
+import com.tubitv.demo.databinding.ExampleUiControlBinding;
+import com.tubitv.media.bindings.UserController;
+import com.tubitv.media.utilities.ExoPlayerLogger;
+
+public class PlayerControllerUI extends FrameLayout implements View.OnClickListener {
+
+    private static final String TAG = PlayerControllerUI.class.getSimpleName();
+
+    private UserController userController;
+
+    private boolean playOrPause = true;
+
+    ExampleUiControlBinding binding;
+
+    public PlayerControllerUI(final Context context) {
+        this(context, null);
+    }
+
+    public PlayerControllerUI(final Context context, @Nullable final AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public PlayerControllerUI(final Context context, @Nullable final AttributeSet attrs, final int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        initLayout(context);
+    }
+
+    public View setController(UserController controller) {
+        this.userController = controller;
+        binding.setController(userController);
+        return this;
+    }
+
+    @Override public void onClick(final View v) {
+        if (userController == null) {
+            return;
+        }
+        switch ((String) v.getTag()) {
+            case "rewind":
+
+                userController.seekBy(-15 * 1000);
+                ExoPlayerLogger.i(TAG, "rewind click");
+                printVideoDetail();
+                break;
+
+            case "play_pause":
+
+                playOrPause = !playOrPause;
+                userController.triggerPlayOrPause(playOrPause);
+
+                ExoPlayerLogger.i(TAG, "play_pause click");
+                printVideoDetail();
+                break;
+
+            case "fastford":
+
+                userController.seekBy(15 * 1000);
+                ExoPlayerLogger.i(TAG, "fastford click");
+                printVideoDetail();
+                break;
+
+            default:
+                return;
+        }
+
+    }
+
+    private void initLayout(Context context) {
```

Comment:
Just making sure we want this private.  Since the layout is named "example_ui_control", it implies it can be overwritten.  Do you want users to overwrite the constructor instead?

Reply:
yes, since we are separating the business logic from UI, now we want to simplify the steps needed to custom the UI. If we allow user to initLayout(), they have to basically understand some of the internal logic, instead of that, we now just let use create their own view with control hooked up, and give it to us. Does it answer your question @sdobek